executor/src/witgen/generator.rs

```diff
@@ -31,15 +36,61 @@ pub struct Generator<'a, T: FieldElement> {
 impl<'a, T: FieldElement> Machine<'a, T> for Generator<'a, T> {
     fn process_plookup<Q: QueryCallback<T>>(
         &mut self,
-        _mutable_state: &mut MutableState<'a, '_, T, Q>,
+        mutable_state: &mut MutableState<'a, '_, T, Q>,
         _kind: IdentityKind,
-        _left: &[AffineExpression<&'a AlgebraicReference, T>],
-        _right: &'a SelectedExpressions<Expression<T>>,
+        left: &[AffineExpression<&'a AlgebraicReference, T>],
+        right: &'a SelectedExpressions<Expression<T>>,
     ) -> Option<EvalResult<'a, T>> {
-        unimplemented!()
+        (right.selector == self.latch).then(|| {
```

Comment:
Could this maybe return early in case the latch does not match?

Reply:
We should really store the relationship between lookup identities and machines, it makes no sense to always check which is the right machine...

Reply:
```suggestion
        if right.selector != self.latch {
            return None;
        } else {
```

Do you mean this? That's equivalent, no?

Reply:
Yes, not too important.

Reply:
> We should really store the relationship between lookup identities and machines, it makes no sense to always check which is the right machine...

Fully agree with that!

I changed the latch check as you suggested.